feat: add ApplyOptions + move to Jest

Favna opened the pull request:
Lets face it, we'll be on our deathbeds by the time Jacz would do this himself.

As for testing the decorator, for the life of me I've tried but I can't wrap my head around all the mocking required just to get it to work. My idea was to have a class which would have a method that returns one of its properties (such as cooldown for a command) but then that requires a KlasaGuild, KlasaMessage, TextChannel, Channel, KlasaClient and god knows what more. That said, if you can come up with a good way to test this decorator with mocks lmk and I'll add it. I couldn't get it to work ¯\\\_(ツ)_/¯

kyranet commented on src/index.ts:
```diff
@@ -14,6 +17,30 @@ export function createMethodDecorator(fn: MethodDecorator) {
+ * Decorator function that applies given options to any Klasa piece
+ * @param options The options to pass to the piece constructor
+ */
+export function ApplyOptions<T extends PieceOptions>(options: T): Function {
```
```suggestion
export function ApplyOptions<T extends PieceOptions>(options: T) {
```

kyranet commented on src/index.ts:
```diff
@@ -14,6 +17,30 @@ export function createMethodDecorator(fn: MethodDecorator) {
+
+/**
+ * Decorator function that applies given options to any Klasa piece
+ * @param options The options to pass to the piece constructor
```
```suggestion
 * @since 0.0.2
 * @param options The options to pass to the piece constructor
```

Favna commented:
Taken from commit body:

- Switched tape for jest. This is primarily because tape is just annoying to work

- Implemented tests for ApplyOptions

- Updated azure pipelines config (ubuntu 18 + jest coverage)

- Moved tests outside of src -> caveat is that there is now a tsconfig.build.json for the building process. __tests__ needs to be included because otherwise VSC is tripping over the decorators but __tests__ should not be in the dist, that's ugly af. This is the best solution.

- Removed npmignore, better to specificy which few files to include rather than which many files to exclude (see files field in package.json)

- Updated eslint config to know we're using Jest

Favna commented:
Looks like GitHub workflows doesn't auto-trigger on this PR 😢
